### PR TITLE
Fixes #31323 - change pulp bulk_load_size to a setting

### PR DESCRIPTION
**Pull request by ianballou**
Changes the `bulk_load_size` to a Setting instead of a SETTING.

### CODE REVIEW

File: app/services/katello/pulp/pulp_content_unit.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def self.unit_handler
         Katello.pulp_server.extensions.send(self.name.demodulize.underscore)
       end
 
-      def self.pulp_units_batch_all(unit_ids = nil, page_size = SETTINGS[:katello][:pulp][:bulk_load_size])
+      def self.pulp_units_batch_all(unit_ids = nil, page_size = ::Setting::Content.find_by(name: "bulk_load_size").value)
```
Comment (jlsherrill) [inline app/services/katello/pulp/pulp_content_unit.rb:49]: Can't you just do  Setting[:bulk_load_size] ?

Comment (ianballou) [reply app/services/katello/pulp/pulp_content_unit.rb:49]: Ah yeah, that should work.

File: app/models/setting/content.rb [hunk 2]
```diff
@@ -150,7 +150,9 @@ def self.default_settings
       self.set('autosearch_while_typing', N_('For pages that support it, automatically perform search while typing in search input.'),
                true, N_('Autosearch')),
       self.set('autosearch_delay', N_('If Autosearch is enabled, delay in milliseconds before executing searches while typing.'),
-               500, N_('Autosearch delay'))
+               500, N_('Autosearch delay')),
+      self.set('bulk_load_size', N_('The amount of items fetched from a single paged Pulp API call.'), 100,
```
Comment (jeremylenz) [inline app/models/setting/content.rb:154]:
```suggestion
      self.set('bulk_load_size', N_('The number of items fetched from a single paged Pulp API call.'), 100,
```
what are the units?

**Review by jeremylenz — COMMENTED** [review]
Does this require an installer change?

File: app/models/setting/content.rb (hunk 2 above)
Comment (ianballou) [reply app/models/setting/content.rb:154]: I'm not sure what to call the units besides items from the API response.

**Comment by ianballou** [comment]
> Does this require an installer change?

Looks like this doesn't require an installer change. We never added `bulk_load_size` support in puppet-katello.

File: app/models/setting/content.rb (hunk 2 above)
Comment (jeremylenz) [reply app/models/setting/content.rb]: Oh, I got it now :)  Yeah I think "number of items" is clear enough then.

**Review by jlsherrill — APPROVED** [review]

**Comment by ianballou** [comment]
@jlsherrill I changed the default bulk_load_size to 2000 from 100.

**Comment by jlsherrill** [comment]
oh thank you, that would have been bad :)

**Pull request merged**